### PR TITLE
Generate YAML files which the user can install in their own cluster

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Generate installable yaml files using a catalog and a profile name.

### Checklist
- [x] Added tests that cover your change
- [x] Added/modified documentation as required (such as the `README.md`)
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] Added a `kind` label to the PR (e.g. `kind/feature`)

### TODO

Figuring out Ownership.

### CODE REVIEW

File: cmd/pctl/main.go [hunk 1]
```diff
@@ -84,6 +88,65 @@ func showCmd() *cli.Command {
 	}
 }
 
+func installCmd() *cli.Command {
+	return &cli.Command{
+		Name:      "install",
+		Usage:     "generate configuration objects for later cluster setup",
```
Comment (aclevername) [inline cmd/pctl/main.go:94]: this doesn't seem right. Its more like "generate resources for installing a profile" or something along those lines

Comment (Skarlso) [reply cmd/pctl/main.go:94]: Yeah, I meant something along the way of saying, setup your cluster, like, your kube cluster, set it up with the profile you will be installing. :D It sounds weird, now reading it after the weekend. :D

File: cmd/pctl/main.go [hunk 2]
```diff
@@ -114,3 +177,30 @@ func printProfile(profile profilesv1.ProfileDescription) error {
...
+		}
+		return nil
+	}
+	for i, a := range artifacts {
+		if err := generateOutput(i, a); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```
Comment (aclevername) [inline cmd/pctl/main.go:206]: does this logic belong in `main`? Makes it harder to unit test

File: pkg/catalog/install.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		}
+	}()
+
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, fmt.Errorf("unable to find profile `%s` in catalog `%s`", profileName, catalogName)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to fetch profile: status code %d", resp.StatusCode)
+	}
```
Comment (aclevername) [inline pkg/catalog/install.go:46]:
```suggestion
	if resp.StatusCode != http.StatusOK {
	        if resp.StatusCode != http.StatusOK {
		     return nil, fmt.Errorf("unable to find profile `%s` in catalog `%s`", profileName, catalogName)
		}
		return nil, fmt.Errorf("failed to fetch profile: status code %d", resp.StatusCode)
	}

```

Comment (aclevername) [inline pkg/catalog/install.go:46]:
```suggestion
	if resp.StatusCode != http.StatusOK{
	        if resp.StatusCode == http.StatusNotFound {
		     return nil, fmt.Errorf("unable to find profile `%s` in catalog `%s`", profileName, catalogName)
		}
		return nil, fmt.Errorf("failed to fetch profile: status code %d", resp.StatusCode)
	}
```

File: pkg/catalog/install_test.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	var (
+		fakeHTTPClient *fakes.FakeHTTPClient
+	)
+
+	BeforeEach(func() {
+		fakeHTTPClient = new(fakes.FakeHTTPClient)
+		catalog.SetHTTPClient(fakeHTTPClient)
+	})
+
+	When("there is an existing catalog and user calls install for a profile", func() {
```
Comment (aclevername) [inline pkg/catalog/install_test.go:31]: can you add some tests for the more common error paths :smile:

File: cmd/pctl/main.go [hunk 5]
```diff
@@ -84,6 +88,65 @@ func showCmd() *cli.Command {
...
+			profilePath, catalogURL, err := parseArgs(c)
+			if err != nil {
+				_ = cli.ShowCommandHelp(c, "install")
+				return err
+			}
+
+			branch := c.String("branch")
+			subName := c.String("subscription-name")
+			namespace := c.String("namespace")
+			configValues := c.String("config-map-name")
```
Comment (aclevername) [inline cmd/pctl/main.go:131]: need to check these aren't empty? Or do we get that for free from the CLI package

Comment (Skarlso) [reply cmd/pctl/main.go:131]: They will have the default value.

File: pkg/catalog/install_test.go (hunk 4 above)
Comment (Skarlso) [reply pkg/catalog/install_test.go:31]: Yep! 👍

File: cmd/pctl/main.go (hunk 2 above)
Comment (Skarlso) [reply cmd/pctl/main.go:206]: Yeah, no. I need to extract this and write some nice tests for it somewhere. :) You're right.

**Comment by Skarlso** [comment]
This will be re-written as per design decisions. So I'm closing this PR and will open another one with different functionality.

**Pull request closed without merging**